syspage.c

```diff
@@ -346,25 +346,25 @@ int syspage_save(void)
 	syspage_common.argCnt++; /* The last char is '\0' */
 	syspage_common.syspage->arg = (char *)((void *)syspage_common.syspage + syspage_common.syspage->syspagesz);
 	syspage_common.syspage->syspagesz += syspage_common.argCnt;
-	hal_memcpy((void *)(syspage_common.syspage->arg), syspage_common.args, syspage_common.argCnt);
+	hal_memcpy(syspage_common.syspage->arg, syspage_common.args, syspage_common.argCnt);
 
 	/* Save programs */
 	syspage_common.syspage->progs = (syspage_program_t *)((void *)syspage_common.syspage + syspage_common.syspage->syspagesz);
 	syspage_common.syspage->progssz = syspage_common.progsCnt;
 	syspage_common.syspage->syspagesz += (syspage_common.progsCnt * sizeof(syspage_program_t));
-	hal_memcpy((void *)(syspage_common.syspage->progs), syspage_common.progs, syspage_common.progsCnt * sizeof(syspage_program_t));
+	hal_memcpy(syspage_common.syspage->progs, syspage_common.progs, syspage_common.progsCnt * sizeof(syspage_program_t));
 
 	/* Save maps */
 	syspage_common.syspage->maps = (syspage_map_t *)((void *)syspage_common.syspage + syspage_common.syspage->syspagesz);
 	for (i = 0; i < syspage_common.mapsCnt; ++i)
-		hal_memcpy((void *)((void *)syspage_common.syspage->maps + i * sizeof(syspage_map_t)), &syspage_common.maps[i].map, sizeof(syspage_map_t));
+		hal_memcpy(syspage_common.syspage->maps + i * sizeof(syspage_map_t), &syspage_common.maps[i].map, sizeof(syspage_map_t));
```

Comment:
That's not good, when you add to pointer of syspage_map_t  * type, you add i * sizeof(syspage_map_t ). So you do here:
syspage_common.syspage->maps + (i * sizeof(syspage_map_t) * sizeof(syspage_map_t).

Use (char *) for pointer arithmetic or just do:
syspage_common.syspage->maps + i or &syspage_common.syspage->maps[i]

Reply:
Thanks for spotting this bug - it's fixed now.